### PR TITLE
doc updates for TorchScript

### PR DESCRIPTION
**Pull request by suo**
Some batched updates:
1. bool is a type now
2. Early returns are allowed now
3. The beginning of an FAQ section with some guidance on the best way to do GPU training + CPU inference

### CODE REVIEW

**Review by jamesr66a — APPROVED** [review]
nice

File: docs/source/jit.rst [hunk 1]
```diff
@@ -493,9 +496,10 @@ Return
     ``return a, b``
 
     .. note::
-        there must be a return statement as the last member of the function
-        and return statements cannot appear anywhere else in the function. This
-        restriction will be removed in the future.
+        TorchScript allows returns in the following circumstances:
+           1. At the end of a function
+           2. In an if-statement where <true> and <false> both return
```
Comment (eellison) [inline docs/source/jit.rst]: It might be worth specifying that we don't allow early returns in the body of loops

Comment (suo) [reply docs/source/jit.rst]: Well, it's probably easier to specify where it *is* allowed, since there are lots of places where it's not.

Comment (eellison) [reply docs/source/jit.rst]: Reading this it's not clear to me that something like

while cond:
     if early return:
         return hi

wouldn't work

**Pull request closed without merging**